test/cql-pytest/test_secondary_index.py

```diff
@@ -100,3 +100,36 @@ def index_used(query, index_name):
         # Local indexes are still preferred over global ones, if they can be used
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v1 = 1 and v3 = 2 and v2 = 2 allow filtering", "my_v2_idx")
         index_used(f"SELECT * FROM {table} WHERE p = 1 and v2 = 1 and v1 = 2 allow filtering", "my_v2_idx")
+
+# Indexes can be created without an explicit name, in which case a default name is chosen.
+# However, due to #8620 it was possible to break the index creation mechanism by creating
+# a properly named regular table, which conflicts with the generated index name.
+# A similar issue still exists for named indexes, so both cases are checked in this scenario.
```

Comment:
If you had split this into two tests, at least one of them (the one where the index creation is expected to succeed) would not need to be scylla_only.